tls/s2n_resume.h

```diff
@@ -91,6 +93,10 @@ typedef enum {
     S2N_SERIALIZED_FORMAT_TLS12_V3,
 } s2n_serial_format_version;
 
+typedef enum {
+    S2N_PRE_ENCRYPTED_STATE_V1 = 1,
+} s2n_pre_encrypted_state;
```

Comment:
To answer your question, I don't see a new API needed for this in the future. It's purely so that going forward we can parse both old and new ticket schemas.

For deployed resumption users deploying a change, instead of a 50% drop in resumption, the drop is only 25% since new code can parse both old and new versions. We still will see a drop though, because old code can't parse new versions.